### ISSUE TITLE
Importing not working in DT 3.0

### ISSUE THREAD

**Issue by muenznej** [darktable-org/darktable#3924]
Hello,
After the update from 2.6 to 3.0 i cannot import images via cable from my Olympus E-M 10 Mark II. The images are "imported" but corrupted. The focal length is f/0.0 and no preview is available. I only have raws to import.They can be previewed with the "import images from camera" gui. 
![image](https://user-images.githubusercontent.com/26899110/71556737-545ce580-2a3c-11ea-8972-d0434719be21.png)
![dt3_error_imprt](https://user-images.githubusercontent.com/26899110/71556739-62ab0180-2a3c-11ea-9763-c2b4834053ba.PNG)
![image](https://user-images.githubusercontent.com/26899110/71556769-bd445d80-2a3c-11ea-80d9-263f7a55a8e0.png)

The files are not readable by other tools after importing.  The files are not corrupted on the sdcard.

Thanks for the wonderful DT and hopefully for a quick solution :)

**Platform (please complete the following information):**
 - Darktable Version: [3.00] 
 - OS: [Windows 8.1]
 - OpenCL activated
 - HD5500 and geforce 940m

**Comment by MStraeten** [comment]
What happens if you import the files after copying themn manually to your disc? Just to reduce the process chain, where errors might occur

**Comment by muenznej** [comment]
importing via the direct camera connection is the only problem. I copied all files to a folder manually and imported that folder which worked out fine.

**Comment by Lucki** [comment]
I can confirm this issue. While direct wire importing worked in Darktable 2.6 it doesn't work anymore in version 3.0. Manually copying the pictures to the hard drive and importing them as a folder still works normally.

In the import dialog Darktable is able to see all pictures. After pressing import Darktable processes all of the selected pictures but every single one is corrupted and displayed as a skull and don't have any correct meta data. Closing and reopening Darktable automatically removes all images but one from the set.
All files and the corresponding `*.xmp` files are created on the hard drive and the file sizes _looks_ normal. However I couldn't verify if they're actually corrupted.

* Darktable 3.0.0
* Windows 10
* Nikon D3500 through [WinUSB](https://www.darktable.org/about/faq/#faq-tethering)
* Radeon Vega 8 and GeForce GT 710

**Comment by Lucki** [comment]
A bit more information gathered through testing:
* The problem was introduced between [2.6.2 and 2.6.3](https://github.com/darktable-org/darktable/compare/release-2.6.2...release-2.6.3)
* I was able to build this on windows beginning with commit 37cba11 which also was unable to import the images correctly.
* Unsurprisingly I wasn't able to build anything before that since the commit fixed some build problems.

So here's a short timeline:
2.6.2 (download, no issue) … 37cba11 (**own build**, has the issue) … 2.6.3 (download, has the issue)

**Comment by Lucki** [comment]
"Activity"

**Comment by houz** [comment]
- How big are the files copied over by darktable and how big are they supposed to be? (`ls -l` will tell you)
- What does `file corrupted_file.raw` (with the actual filename of course) tell you?
- Do you see any related output when starting darktable from a terminal and then try to copy files from the camera?

**Comment by Lucki** [comment]
~~~
$ ls -l
-rw-r--r-- 1 lucki lucki 17148553 16. Mär 2020   corrupted_import.NEF
-rw-r--r-- 1 lucki lucki 17129543 16. Mär 2020   original_image.NEF
~~~

~~~
$ file corrupted_import.NEF original_image.NEF 
corrupted_import.NEF: TIFF image data, little-endian, direntries=28, height=120, bps=352, compression=none, PhotometricIntepretation=RGB, manufacturer=, model=, orientation=upper-left, width=160
original_image.NEF:   TIFF image data, little-endian, direntries=28, height=120, bps=352, compression=none, PhotometricIntepretation=RGB, manufacturer=NIKON CORPORATION, model=NIKON D3500, orientation=upper-left, width=160
~~~

<details><summary>I get the following lines after closing Darktable in the Msys2 terminal, they're not printed live in the terminal:</summary>

~~~
$ /opt/darktable/bin/darktable.exe
[rawspeed] (20200316_0001.NEF) virtual std::unique_ptr<rawspeed::RawDecoder> rawspeed::RawParser::getDecoder(const rawspeed::CameraMetaData*), line 91: No decoder found. Sorry.
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 0 to 2097152
[import_session] Using filename 20200316_0001.NEF.
Magick: Unrecognized image format () [No such file or directory].
[dt_imageio_large_thumbnail GM] thumbnail not found?
allocation failed???
~~~

**Comment by hanosteyn** [comment]
Hello
I am new to Darktable and have only recently looked at importing my files from the camera to the file structure on my harddrive.

It seems that I am also getting the same problem of the file importing, being correctly renamed and copied to the correct place, but then the file is corrupt and not usable.  This also seems to be a problem in tether mode.  The camera connects fine, but the downloaded image is corrupt.

I have tried the import with digiKam and that works fine.
 
Is there a solution to this problem?

Thanx very much

My setup is:
Darktable 3.0.2
Windows 10
Nikon D5600
WinUSB

**Comment by Lucki** [comment]
While using an old (20181211) [Msys2 installation](http://repo.msys2.org/distrib/x86_64/) I can build the current master without problems by making sure I'm not updating the database with `y`.
The resulting build of Darktable has the import **working**.

I've used the other up-to-date Msys2 installation to also build the current master.
The resulting build of Darktable has the import **corrupted**.

To test which file is causing the failing import I've copied one file at a time into the working build from the failing build and stopped at the first file with which Darktable is unable to import the image correctly. While with some files Darktable wasn't able to start anymore (`libdarktable.dll`, `libexiv2.dll`) I have found that replacing `libglib-2.0-0.dll` is causing a previously working Darktable build to fail the import.
Unfortunately I'm unable to test this the other way around. By replacing the file with the file build in the old building environment Darktable is unable to start.

**Comment by fstokke** [comment]
I have the same problem (Darktable 3.0.2 on Windows 10). I had to use Zadig to install the WinUSB driver to get Darktable to recognize the camera - out of the box installation did not work. It now recognizes the camera and import runs, but imported image is corrupt (cannot open the imported jpg from file system in other programs). Images on SD card are OK, and can be opened in other programs.

**Comment by nate911** [comment]
Same issue here. Darktable 3.0.2 on Windows 10 with Nikon D800. Like fstokke mentioned, direct SD import works fine.

**Comment by Lucki** [comment]
Now with 3.2.1 released I thought it would be easy to recompile and get the new version but no, there was a commit in between my last comment and the release of 3.2.1 which triggers a compiler bug on the old gcc. Luckily it worked by just updating gcc and it still has the import working. So here's a step-by-step guide because this was probably my last compilation of darktable:

* Instead of the current Msys2 release install an older version like (20181211) from here: http://repo.msys2.org/distrib/x86_64/
* Follow the installation guide A) and install all dependencies but make sure you're not gonna update the package database by using `y` in combination with `pacman`. This ensures you're getting all the packages from the time the package database was at (2018).
* optional: use `git checkout release-3.2.1 && git submodule update` to build at a release instead of the current development version
* Make sure you have all dependencies by now, the `cmake` commands will tell you if something is missing.
* When you're at the point of running `cmake` after cloning we have to update gcc in order to be able to build successfully
* [We need new keys](https://www.msys2.org/news/#2020-06-29-new-packagers):
  * `curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz`
  * `curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz.sig`
  * `pacman -U msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz`
* Update the database `pacman -Sy`
* Update gcc and it's dependencies: `pacman -S mingw-w64-x86_64-gcc mingw-w64-x86_64-[…]` (there are six in total)
* Now proceed with the `cmake` commands in the installation guide

Following the steps above I'm able to get a recent Darktable with the import **working**. But considering the hassle and the lacking support from the developers I won't build any further daktable releases.

Here's the installer, I hope it works for you:
https://github.com/Lucki/darktable/releases/download/release-3.2.1/darktable-3.2.1-win64.exe

Some advises:
* The tooltips in the lighttable cause hell of a flickering so turn them off at the star in the top right
* Make sure the import works before updating to further releases

**Comment by MStraeten** [comment]
why not contribute to the project as a maintainer for windows? It’s not fair to blame the developers since most - if not all - of them don’t use Windows for development. So they even cant reproduce windows specific bugs especially when these are originated in the msys/mingw environment.

**Comment by TurboGit** [comment]
> Following the steps above I'm able to get a recent Darktable with the import **working**. But considering the hassle and the lacking support from the developers I won't build any further daktable releases.

If all devs were thinking alike darktable won't even exists. But this kind of thinking seems common from Windows devs, that's why the Windows support came so late. Remember there is a single Windows devs that talked to us to maintain this port. And actually Peter is looking for help, so maybe you can contact him and team together?

**Comment by Lucki** [comment]
Now there's suddenly activity…

> why not contribute to the project as a maintainer for windows? It’s not fair to blame the developers since most - if not all - of them don’t use Windows for development.

Because I'm neither a developer nor using windows. All I've done is building the recent release with a two year old system - the last known working state - on machine that's not even mine. That's not maintaining, that's a workaround.
I've answered all questions from maintainers, I've tried around myself and provided a step-by-step guide on how to achieve the same *workaround*. That's as far as I can go without any feedback or questions from the maintainers.

> > Following the steps above I'm able to get a recent Darktable with the import **working**. But considering the hassle and the lacking support from the developers I won't build any further daktable releases.
> 
> If all devs were thinking alike darktable won't even exists. But this kind of thinking seems common from Windows devs, that's why the Windows support came so late.

I am no windows dev, not even a dev. I've set the whole build environment up on a foreign pc. That's why I don't want to do this on further releases, dunno why other windows dev think alike, I always thought windows devs have own windows PCs…

> Peter is looking for help, so maybe you can contact him and team together?

If I actually would have a windows system and knowledge about windows in any way yes, but I don't have them. I've just followed the build instructions and [even corrected some of them](https://github.com/darktable-org/darktable/commit/57e69b70e437b465f4a4811676800e4b3c72e503#diff-9ee5be0562561d51ef3f5047a231a40b).

**Comment by Lucki** [comment]
Don't use the Darktable I've provided above in a productive environment. It seems to corrupt the file system on importing so **using this is will result on data loss**.

**Comment by yzoned** [comment]
This bug stopped my workflow with Darktable, corrupt images from camera import. Was hoping to replace Lightroom with Darktable but it seems this bug is not getting fixed.

**Comment by ptilopteri** [comment]
upgrade darktable to 3.2.1 or master from git and see if it helps

**Comment by muenznej** [comment]
Hello again, just wanted to report, that this issue is still present and also happens with a Canon 5D mark ii. With that camera its really a pain, since i cannot really import images after using the zadig device driver. The problem is identical to that of my olympus camera: all files are corrupt. Since that camera supports tethering I can also say that everything but file transfer works fine. So camera control is correctly working, but the image is not imported correctly.
In this current form i cannot use the tethering which is a bit sad. I ll try to rewind the drivers until maybe the issue will be resolved. Since this is almost a year now since I reported it i am a bit sad tbh :(

**Comment by muenznej** [comment]
oh thanks i ll give it a try and report back :D

**Comment by ogloke** [comment]
> 
> 
> Hello again,
> just wanted to report, that this issue is still present and also happens with a Canon 5D mark ii.
> 
> With that camera its really a pain, since i cannot really import images after using the zadig device driver. The problem is identical to that of my olympus camera: all files are corrupt.
> 
> Since that camera supports tethering I can also say that everything but file transfer works fine. So camera control is correctly working, but the image is not imported correctly.
> 
> In this current form i cannot use the tethering which is a bit sad. I ll try to rewind the drivers until maybe the issue will be resolved. Since this is almost a year now since I reported it i am a bit sad tbh :(
> 
> P.S.: I am on darktable 3.2.1

I have the exact same problems with a Canon 5D. I can use a card reader to get my images but the loss of tethered shooting is a pain as darktable was the only software that would work with the 5D

**Comment by Getondown** [comment]
Thanks @muenznej , I am experiencing the same problem as well. It looks like you are our last hope. I seem to recall it working earlier in the year so maybe it started with the upgrade to vers 3.2.1. I was probably using an older version at the time. I wonder if I can downgrade to an earlier version.

**Comment by ptilopteri** [comment]
@Getondown 
if you downgrade, you will need to replace your library.db with one from the same
time period to downgrade to.  and perhaps darktablerc

or you might (re)move your ~/.config/darktable
and reimport your images.

and possibly the ~/.cache/darktable/mipmaps* will need to be replaced.

downgrading is not a supported path.

**Comment by Getondown** [comment]
I didn't know that. Thanks for the information.

**Comment by Getondown** [comment]
There is a post here: https://discuss.pixls.us/t/please-help-darktable-tethering-issues-with-canon/12639/16 from 2019 that deals with the darktable issue. The version of darktable was 2.6.2. Supposedly the issue was resolved by making a change to the **Base Directory Naming Pattern** under **Session Options**. The naming pattern "**$(PICTURES_FOLDER)/Darktable**" was replaced by the file path, "**C:\Users ....**" and the suggestion to try either using a "\\"  or "/".

The person who asked the question says it worked for them. I don't understand why it would work and could not get it to work for me. I left a note on the site asking for more information on what was done but since it an old post, I don't know if or when I might get a response. Any thoughts?

**Comment by johnny-bit** [comment]
@Getondown - both users in that thread were last seen on forums in last year so it might be hard to get them to reply ;)

So to explain: user had "C:\Users\jsimp\Desktop" and chenged to either "C:\\Users\\jsimp\\Desktop" or "C:/Users/jsimp/Desktop" - try either of those ;)

**Comment by Getondown** [comment]
Darktable asks for the naming pattern not the full path so it is strange that adding the full path will make it work. Nevertheless, I have tried several variations as suggested without success. I will try again.

I notice that the location that darktable looks for the photo (see **Full Path** under **Image Information**)  is "C:\Users\MyName\Pictures/Darktable$(YEAR) etc"

Should that be "C:\Users\MyName\Pictures\Darktable$(YEAR) etc? In other words, should the forward slash in front of darktable be a back slash? I assume that with a forward slash, after importing the photo, it would not be able to save it on Windows.

**Comment by johnny-bit** [comment]
This depends on msys2 handling inside windows since windows decided to be "cute" and decided that path separator being same as well-established escape character is great idea ;)

Coding snark aside - with `\` character try `\\` and with `/` character - it should be autotranslated to "understandable" path.

**Comment by Getondown** [comment]
I tried making modifications to the original naming pattern **"$(PICTURES_FOLDER)/Darktable"** by changing the forward slash.  This is how the file path changed with each modification.

/ : Darktable path is C:\Users\MyName\Pictures/Darktable$(YEAR) etc
// : Darktable path is C:\Users\MyName\Pictures//Darktable$(YEAR) etc
\ : Darktable path is C:\Users\MyName\PicturesDarktable$(YEAR) etc  (no slash between Pitures and Darktable)
 \ \ : Darktable path is C:\Users\MyName\Pictures\Darktable$(YEAR) etc

So it would seem that using \ \ , points Darktable to the correct path for the photos (although it seems that regardless of the type of slash, the files are saved). Anyway, the files still do not show up properly on the screen with a \ \ . 

When I go to the file location on my hard drive, the files are there but when I try to open them with an image viewer, it says that the files are either damaged, corrupted or not supported. The files on the SDcard are fine. I can read them. Somehow, Darktable is causing them to be corrupted when tethering.

**Comment by johnny-bit** [comment]
damn... remove `$(YEAR)` from the path?

also - what version do you use and with what camera do you have problems? Olympus like @muenznej or Canon or Nikon? Also - do you have any option to try darktable on linux? Because for me tethering works wonderfully (Canon 77D)

**Comment by Getondown** [comment]
When I removed the $(YEAR) from the Subdirectory Naming Pattern, the results were the same as before except that the file naming pattern did not indicate the year. I am using Darktable 3.2.1.

I think this problem is more than a year old. I am using a Canon EOS 1000D. I did get it to work with an older version of Darktable; sometime in 2019, I think. As far as I know it is only a Windows problem. 

I have linux installed as well but I have not used it for a long time. I don't remember offhand how to boot it up instead of Windows but I should be able to figure it out. Still, since I am on Windows all the time, using it on Linux would severly limit how often I can use it.

**Comment by johnny-bit** [comment]
Yeah, I agree... There were couple of fixes to import functionality by @jenshannoschwalm that will make it into 3.4 so maybe that version will work better for you? Let's wait and see.

**Comment by Getondown** [comment]
Let's hope so. A guy on reddit had a similar problem. I asked him whether he was able to get tethering to work in darktable. He said  "I ended up writing my own program using libgphoto2 for USB bound control of a digital camera."

**Comment by ralfbrown** [comment]
> This depends on msys2 handling inside windows since windows decided to be "cute" and decided that path separator being same as well-established escape character is great idea ;)
> 
> Coding snark aside - with `\` character try `\\` and with `/` character - it should be autotranslated to "understandable" path.

Well, you can sorta blame 1970's CP/M for the backslashes, but the Windows API calls themselves will accept either forward or backslash as directory separator.

As for the backslashes themselves, when MS-DOS 2.0 added subdirectories it used them instead of forward slash because forward slash was the character starting commandline flags in MS-DOS 1.x, which in turn copied the convention from CP/M (since MS-DOS 1.0 was basically an 8086 clone of the 8080-based CP/M).  But even back then, the OS API calls would accept either, and a program was free to use dash instead of slash as the flag character.  There was even an API call to ask what the flag character should be, so that programs could adapt, but it never caught on.  MS-DOS 2.x had a CONFIG.SYS option to set the flag character, but it was quietly dropped in 3.0 (the API call was also dropped, and then reinstated in 4.0, but it always returns '/' now).

**Comment by Getondown** [comment]
A Merry Xmas to you all. I just downloaded and installed Darktable version 3.4 and the issue still exists.

**Comment by johnny-bit** [comment]
@muenznej - does the issue persist for you too?

**Comment by muenznej** [comment]
Hi there and happy holidays to you.
I just recently (today) switched to DT 3.4 on LinuxMint and I could not really test much, but i had one of those "no error message" crashes. Let my try to elaborate what happend:
I tried to import from a folder (~4000 images) and it became unresponsive ( maybe it was just working and i could have waited) but i restarted DT and tried importing again. And then DT crashed without a notification or error log, like i explained in #6645.
I am not working with windows much any more, but i am pretty convinced that those two issues are caused by the same "bug".

In #6645 it was suggested that i should try to run darktable with a custom/dummy configdir and this works ( at least as far as i can tell). So what i am currently trying is backing up that folder on every "proper" import and replace it whenever this "import bug"  prevents me from using DT.

I assume something is happening to the database files during a improper import that causes a crash whenever this path is imported again. Its a bit of a gut feeling since this bug happens only once in a while, but a fresh configdir resolves this problem totally.

sorry for the lack of conclusive input, but my knowledge and time are very limited :/

**Comment by johnny-bit** [comment]
Since you are on linux now it's a bit easier to debug... run darktable with `-d all` in terminal and then at the point of crash you'll have good debug data :)

**Comment by muenznej** [comment]
> Since you are on linux now it's a bit easier to debug... run darktable with `-d all` in terminal and then at the point of crash you'll have good debug data :)

aye sir, i  ll run it with this paramter from now on. thnx

**Comment by muenznej** [comment]
can i dump that debug info in a file? and if yes how?

**Comment by johnny-bit** [comment]
run darktable as eg `darktable -d all > darktable.log`. should be fine :)

**Comment by JennaSys** [comment]
I installed 3.4.1 on a Windows 10 PC and ran into the same problems as others here tethering to an older Canon XTi. First with the USB driver, which updating with Zadig per the FAQ fixed.  Then once the camera was being detected, had the issue with image corruption.  Neither raw files not JPGs were readable in darktable or other software.  I tried making changes to the file naming parameters with no luck.  I uninstalled 3.4.1 and installed 2.6.2 as someone else here mentioned, and after a reboot everything was working fine in the older version.

Also for reference, I have previously been using the latest version of darktable with the same camera on Linux Mint without any problems.

**Comment by Nilvus** [comment]
darktable 3.6 will be out in a week and will also add a whole new import module. @muenznej and @JennaSys: it would be good if you can test again if you still reproduce this issue in darktable 3.6 in July and report back here what you have. If you don't reproduce it on 3.6, @muenznej consider closing this issue (or ask for it).

**Comment by Nilvus** [comment]
I close as no answer and import feature had been completely rewritten. If needed a new issue could be open if this occurs in 3.6.
